### ISSUE TITLE
Async tasks

### ISSUE THREAD

**Issue by waj** [getsentry/sentry-elixir#172]
Currently `Sentry.Client` makes use of `Task.Supervisor.async_nolink` to dispatch the events. However this will send a message back to the process sending the event with the result once delivered. This might be an issue when the caller is not expecting the async result.

For example, I found this issue when a GenServer sends a Sentry message, but the `handle_info` implementation doesn't consider arbitrary messages. It crashes with a message like this one:
```
GenServer MyGenServer terminating
** (FunctionClauseError) no function clause matching in MyGenServer.handle_info/3
    lib/my_gen_server.ex:31: MyGenServer.handle_info({#Reference<0.0.1.153327>, {:ok, "bcc1234d0ece458aa3da252caa077d62"}}, nil, #<DateTime(2017-06-14T12:00:54.396570Z Etc/UTC)>)
```

Also, if the sending process never reads the inbox, the responses will pile up forever.

I tried change the call to `Task.Supervisor.start_child` but that breaks some of the tests, and I'm not really sure what's the real solution here.

**Comment by mitchellhenke** [comment]
Hey, thanks for opening this!

This is hard because `async_nolink` allows users to read the result of the task to get the Sentry Event ID or see if there was an error sending the event.  `start_child` doesn't allow that.

The docs for sentry-elixir around this issue will be updated to make people aware, but I'm not sure on the best path forward as we have to allow for checking the result.

It's worth mentioning that a custom client module could be used in place of the default one, and it could be the exact same, but using `start_child` instead of `async_nolink`.  The only test that fails is one that tests the config testing mix task.  No other internals depend on it, so it _shouldn't_ break anything.

If you have any thoughts on a solution, I'd be happy to hear them.

**Comment by waj** [comment]
I can see three different scenarios:
  * Receive the result asynchronously (using either `Task.await` or by receiving the message)
  * Receive the result synchronously (block the caller)
  * Ignore the result

Maybe the `Sentry` APIs could receive an optional argument to specify how to get back the result. How about something like:
```
Sentry.capture_exception(ex, result: :async | :sync | :none)
```

Then I vote to have `:none` as default :-)

**Comment by mitchellhenke** [comment]
@waj That was along the lines of what I was thinking.  I will put together a PR soon!

**Comment by mitchellhenke** [comment]
closed in #174, but I decided to keep `:async` as the default for now.
